src/Polly.Core/Registry/ResiliencePipelineRegistry.cs

```diff
@@ -122,20 +122,16 @@ public ResiliencePipeline GetOrAddPipeline(TKey key, Action<ResiliencePipelineBu
             return pipeline;
         }
 
-        return _pipelines.GetOrAdd(key, k =>
-        {
-            var componentBuilder = new RegistryPipelineComponentBuilder<ResiliencePipelineBuilder, TKey>(
-                _activator,
-                k,
-                _builderNameFormatter(k),
-                _instanceNameFormatter?.Invoke(k),
-                configure)
-            ;
-
-            (var contextPool, var component) = componentBuilder.CreateComponent();
-
-            return new ResiliencePipeline(component, DisposeBehavior.Reject, contextPool);
-        });
+        var componentBuilder = new RegistryPipelineComponentBuilder<ResiliencePipelineBuilder, TKey>(
+            _activator,
+            key,
+            _builderNameFormatter(key),
+            _instanceNameFormatter?.Invoke(key),
+            configure);
+
+        (var contextPool, var component) = componentBuilder.CreateComponent();
+
+        return _pipelines.GetOrAdd(key, new ResiliencePipeline(component, DisposeBehavior.Reject, contextPool));
```

Comment:
Similar question here - this doesn't seem better at first glance as we're always creating the `componentBuilder` even if it doesn't need to be used.

Reply:
This always avoids the closure and delegate allocation, but only allocates a `componentBuilder` if it wasn't in the dictionary already (`TryGetValue` call above checks for that).